### PR TITLE
feat(botocore): support eventbridge

### PR DESCRIPTION
**Pull request by idonava**

### CODE REVIEW

File: epsagon/events/botocore.py [hunk 1]
```diff
@@ -117,13 +117,82 @@ def update_response(self, response):
...
+        :param start_time: Start timestamp (epoch)
+        :param response: response data
+        :param exception: Exception (if happened)
+        """
+
+        super(BotocoreCloudWatchEvent, self).__init__(
+            wrapped, instance, args, kwargs, start_time, response, exception
+        )
+        _, request_data = args
+        entries = request_data.get('Entries', [{}])[0]
```
Comment (maorlx) [inline epsagon/events/botocore.py:155]: I would change it so you initially get 'Entries' and then check it (so it won't explode if request_data.get('Entries') is empty)

File: epsagon/events/botocore.py [hunk 2]
```diff
@@ -117,13 +117,82 @@ def update_response(self, response):
...
+        self.resource['name'] = (
```
Comment (maorlx) [inline epsagon/events/botocore.py:156]: can do entries.get('EventBusName', 'Cloudwatch Events')

File: epsagon/events/botocore.py [hunk 3]
```diff
@@ -117,13 +117,82 @@ def update_response(self, response):
...
             response['ResponseMetadata']['HTTPStatusCode']
 
 
+class BotocoreCloudWatchEvent(BotocoreEvent):
+    """
+    Represents cloudwatch events (eventbridge) botocore event.
+    """
+
+    RESOURCE_TYPE = 'eventbridge'
+    RESOURCE_TYPE_UPDATE = 'events'
```
Comment (maorlx) [inline epsagon/events/botocore.py:135]: not important but would rename to event (non plural)

File: epsagon/events/botocore.py [hunk 4]
```diff
@@ -117,13 +117,82 @@ def update_response(self, response):
...
+    def update_response(self, response):
+        """
+        Adds response data to event.
+        :param response: Response from botocore
+        :return: None
+        """
+        super(BotocoreCloudWatchEvent, self).update_response(response)
+        if self.resource['operation'] == 'PutEvents' and 'Entries' in response:
+            self.resource['metadata']['aws.cloudwatch.event_id'] = \
+                response['Entries'][0]['EventId']
```
Comment (maorlx) [inline epsagon/events/botocore.py:194]: it would be good to make sure response['Entries'] is not empty

File: epsagon/triggers/aws_lambda.py [hunk 5]
```diff
@@ -439,7 +439,10 @@ def __init__(self, start_time, event, context):
         super(EventsLambdaTrigger, self).__init__(start_time)
 
         self.event_id = str(event['id'])
-        self.resource['name'] = str(event['resources'][0].split('/')[-1])
+        name = 'CloudWatch Events'
```
Comment (maorlx) [inline epsagon/triggers/aws_lambda.py:442]: in case of events bridge event, what would be the resource name for the event?

**Review by maorlx — CHANGES_REQUESTED** [review]

File: epsagon/events/botocore.py (hunk 3 above)
Comment (idonava) [reply epsagon/events/botocore.py:135]: 'events' is the aws resource name `boto3.client('events')...`

File: epsagon/triggers/aws_lambda.py (hunk 5 above)
Comment (idonava) [reply epsagon/triggers/aws_lambda.py:442]: the trigger resource name will be always 'CloudWatch Events' because EventBusName doesn't exist here.

**Review by maorlx — APPROVED** [review]

**Pull request merged**